keys.hh

```diff
@@ -782,7 +782,7 @@ auto format_pk(const WithSchemaWrapper& pk, FormatContext& ctx) {
         if (utils::utf8::validate((const uint8_t *) key.data(), key.size())) {
             out = fmt::format_to(out, "{}", key);
         } else {
-            out = fmt::format_to(out, "{}", "<non-utf8-key>");
+            out = fmt::format_to(out, "{}", fmt_hex{to_bytes_view(sstring_view(key))});
```

Comment:
Why do we even check the utf-ness of the key's bytes, instead of only printing it as UTF-8 when it has the utf-8 type, and as bytes when it's not a UTF-8 type? Or perhaps even better, for other types also print the key wisely (as integer, or whatever it is)?

Reply:
Good point. We should have smart pretty printers, which don't try to pretty print types which cannot be pretty-printed. Maybe this is already the case, and this is only to deal with corrupt keys?

Reply:
we do print bytes as hex. the underlying implementation uses `to_string_impl_visitor` and dispatch the by the type of the data type. in general, `to_string()` is able to emit readable string. @michoecho probably what we want is to print the ASCII subset of utf-8 strings, and print those with non-utf-8 chars as hex ? as, to render the non-ascii code points properly would require the editor and terminal's settings.